frontend/src/pages/category/CategoryRow.tsx

```diff
@@ -1,48 +1,38 @@
-import React, { useState } from 'react';
 import { CategoryItem } from '../../interfaces/category';
-import EditCategoryModal from '../../components/modals/EditCategoryModal';
-import DeleteCategoryModal from '../../components/modals/DeleteCategoryModal';
-import { EditButton } from '../../components/buttons/EditButton';
-import { DeleteButton } from '../../components/buttons/DeleteButton';
+import ModalWrapper from '../../components/modals/ModalWrapper';
+import { BaseButton } from '../../components/buttons/BaseButton';
+import { useModal } from '../../hooks/useModal';
+import { ModalType } from '../../components/modals/ModalContext';
 
 interface CategoryItemProps {
   category: CategoryItem;
   onEdit: (category: CategoryItem) => void;
   onDelete: (id: string) => void;
 }
 
-const Category: React.FC<CategoryItemProps> = ({ category, onEdit, onDelete }) => {
-  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+const CategoryRow: React.FC<CategoryItemProps> = ({ category, onEdit, onDelete }) => {
+  const { modalType, openModal } = useModal();
 
-  const handleEditModalOpen = () => {
-    setIsEditModalOpen(true);
-  };
-
-  const handleEditModalClose = () => {
-    setIsEditModalOpen(false);
-  };
-
-  const handleDeleteModalOpen = () => {
-    setIsDeleteModalOpen(true);
-  };
-
-  const handleDeleteModalClose = () => {
-    setIsDeleteModalOpen(false);
-  };
   return (
     <div className="w-1/2 m-1 p-3 rounded-lg shadow border border-gray-500 bg-gray-400 dark:bg-gray-800">
       <div className="flex flex-row justify-between items-center">
         <span>{category.name}</span>
         <div className="flex flex-wraps items-center space-x-3">
-          <EditButton onClick={handleEditModalOpen} />
-          <DeleteButton onClick={handleDeleteModalOpen} />
+          <BaseButton
+            onClick={() => openModal(ModalType.EDIT)}
+            className="bg-yellow-500 hover:bg-yellow-600 focus:ring-yellow-700"
+            label="Edit"
+          />
+          <BaseButton
+            onClick={() => openModal(ModalType.DELETE)}
+            className="bg-red-500 hover:bg-red-600 focus:ring-red-700"
+            label="Delete"
+          />
         </div>
-        {isEditModalOpen && <EditCategoryModal category={category} closeModal={handleEditModalClose} onEdit={onEdit} />}
-        {isDeleteModalOpen && <DeleteCategoryModal category={category} closeModal={handleDeleteModalClose} onDelete={onDelete} />}
+        {modalType !== ModalType.NONE && <ModalWrapper category={category} onEdit={onEdit} onDelete={onDelete} />}
```

Comment:
ideally condition that `modalType !== ModalType.NONE` should be handled inside of `ModalWrapper`

Reply:
done